src/components/Layout/PhoneNumberInput.tsx

```diff
@@ -71,14 +85,20 @@ export const PhoneNumberInput: FunctionComponent<{
           style={styles.countryCode}
           keyboardType="phone-pad"
           value={countryCodeValue}
-          onChange={({ nativeEvent: { text } }) => onChangeCountryCode(text)}
+          onChangeText={text => onChangeCountryCode(text)}
         />
         <AppText style={styles.hyphen}>-</AppText>
         <TextInput
           style={styles.numberInput}
           keyboardType="phone-pad"
-          value={mobileNumberValue}
-          onChange={({ nativeEvent: { text } }) => onChangeMobileNumber(text)}
+          value={value}
+          onChangeText={text => {
+            try {
+              const cleanText: string = text.replace(" ", "");
+              phoneNumberFormatted(text);
+              onChangeMobileNumber(cleanText);
+            } catch {}
+          }}
```

Comment:
Perhaps this can be extracted into a function that does the following:

- Formats the phone number
- Prevents input of more than the maximum length
- Allows input of less than 2 without hard coding `length <= 2` (what if the library decides 2 is not short tomorrow?)
- Removes formatting from the formatted phone number and passes it to `onChangeMobileNumber`

```suggestion
          onChangeText={text => onHandleChangeText(text)}
```

Reply:
added onHandleChangeText 
- Formats the phone number
- Input length not restricted as to fit International Numbers next time
-Removed "SG" and used getRegionCodeForCountryCode so can fit formatting for International Numbers

Reply:
@yongjiajie  Will like to check if `LoginMobileNumberCard` from `line 48-50`:
  const [countryCode, setCountryCodeValue] = useState(
    config.fullMobileNumber?.countryCode ?? "+65"
  );

If this is at +65, if I try to type other Country Code & Number in PhoneNumberInput, will the formatting prevent me?

Reply:
No it won't, the value within `useState` is just the initial value.